pkg/kubectl/cmd/diff.go

```diff
@@ -413,29 +340,30 @@ func RunDiff(f cmdutil.Factory, diff *DiffProgram, options *DiffOptions) error {
 		Unstructured().
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, &options.FilenameOptions).
-		Local().
 		Flatten().
 		Do()
 	if err := r.Err(); err != nil {
 		return err
 	}
 
-	dl, err := NewDownloader(f)
-	if err != nil {
-		return err
-	}
-
 	err = r.Visit(func(info *resource.Info, err error) error {
 		if err != nil {
 			return err
 		}
 
-		remote, _ := dl.Download(info)
+		local := info.Object.DeepCopyObject()
+		if err := info.Get(); err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+			info.Object = nil
+		}
+
 		obj := InfoObject{
-			Remote:  remote,
-			Info:    info,
-			Parser:  parser,
-			Encoder: cmdutil.InternalVersionJSONEncoder(),
+			LocalObj: local,
+			Info:     info,
+			Encoder:  cmdutil.InternalVersionJSONEncoder(),
```

Comment:
Please use "scheme.DefaultJSONEncoder()" instead of "InternalVersionJSONEncoder()". This would remove a reference to an internal version of resources. We can safely change this, since the objects are in unstructured, not internal, format.